### ISSUE TITLE
Assertion Error: assert root.size == 1

### ISSUE THREAD

**Issue by ashudeep** [JianGoForIt/YellowFin_Pytorch#8]
I get the following error with the following stack:
`optimizer.step() ` 
`self.after_apply() `
`self.get_mu()`
`assert root.size == 1`

It works fine for most of my runs but fails the assertion sometimes. I will try reproducing it in a simpler setting.

**Comment by ashudeep** [comment]
Didn't notice the closed issue #1.
The error was possibly because of high learning rate.
